File: Exemplos/App_IntentDigitalHUB_Kotlin/Elgin_IntentDigitalHUB_Java-font/app/src/main/java/com/elgin/intent_digital_hub/IntentDigitalHubService/TERMICA/TermicaCommand.kt
package com.elgin.intent_digital_hub.IntentDigitalHubService.TERMICA

import com.elgin.intent_digital_hub.IntentDigitalHubService.IntentDigitalHubCommand
import com.elgin.intent_digital_hub.IntentDigitalHubService.IntentDigitalHubModule
import com.google.gson.annotations.SerializedName
import java.io.Serializable

/**
 * Classe que generaliza todos os comandos do módulo TERMICA, definindo o módulo do comando e o seu tipo de retorno
 */
abstract class TermicaCommand protected constructor(functionName: String?) :
    IntentDigitalHubCommand(functionName, IntentDigitalHubModule.TERMICA) {
    //O retorno dos comandos de impressora é sempre um int
    @SerializedName("resultado")
    private var resultado: Int = 0

    open fun getResultado(): Int {
        return resultado
    }
}

File: Exemplos M10/App_IntentDigitalHUB_ReactNative/Elgin_IntentDigitalHUB_ReactNative-font/src/intentDigitalHubService/TERMICA/TermicaCommand.ts
import {IntentDigitalHubCommand} from '../IntentDigitalHubCommand';
import {IntentDigitalHubModule} from '../IntentDigitalHubModule';

/**
 * Classe que generaliza todos os comandos do módulo TERMICA, definindo o módulo do comando e o seu tipo de retorno
 */
export abstract class TermicaCommand extends IntentDigitalHubCommand {
  constructor(functionName: string) {
    super(functionName, IntentDigitalHubModule.TERMICA);
  }
}
